File: Codes/2244-minimum-rounds-to-complete-all-tasks.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-05-14 11:48:18
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-05-14 11:50:48
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> ma;
        for (int t : tasks) {
            ma[t]++;
        }
        int ans = 0;
        for (auto& [val, times] : ma) {
            if (times == 1) {
                return -1;
            }
            ans += (times + 2) / 3;
        }
        return ans;
    }
};

File: Codes/2244-minimum-rounds-to-complete-all-tasks.py
'''
Author: LetMeFly
Date: 2024-05-14 11:51:16
LastEditors: LetMeFly
LastEditTime: 2024-05-14 11:52:32
'''
from typing import List
from collections import defaultdict

class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        ma = defaultdict(int)
        for t in tasks:
            ma[t] += 1
        ans = 0
        for val, times in ma.items():
            if times == 1:
                return -1
            ans += (times + 2) // 3
        return ans